advanced-javascript/tests/arrays.test.js

```diff
@@ -18,43 +18,60 @@ describe('Arrays', () => {
   describe('`each`', () => {
     it('should be a function', () => {
       const each = arrayFunctions.each;
-      expect(each).to.be.a('string');
+      expect(each).to.be.a('function');
     });
     // begin here
+    it('should invoke a callback arguments', () => {
+      let count = 0;
+      arrayFunctions.each([1, 2, 3], (element) => {
+        count += element;
+      });
+      expect(count).to.equal(6);
+    });
   });
 
   describe('`map`', () => {
     it('should be a function', () => {
       const map = arrayFunctions.map;
-      expect(map).to.be.an('object');
+      expect(map).to.be.an('function');
+    });
+    it('should modify the existing array by the parameter you specify in the item variable', () => {
+      const elements = [1, 2, 3];
+      const mappedArr = arrayFunctions.map(elements, n => (n * 2));
+      expect(mappedArr).to.deep.equal([2, 4, 6]);
     });
   });
 
   describe('`reduce`', () => {
     it('should be a function', () => {
       const reduce = arrayFunctions.reduce;
-      expect(reduce).to.be.a('number');
+      expect(reduce).to.be.a('function');
+    });
+    it('should reduce elements to a single value', () => {
+      const elements = [1, 2, 3];
+      const reduce = arrayFunctions.reduce(elements, (memo, item) => (memo + item))
+      expect(reduce).to.be.equal(6);
     });
   });
 
   describe('`find`', () => {
     it('should be a function', () => {
       const find = arrayFunctions.find;
-      expect(find).to.be.an('array');
+      expect(find).to.be.an('function');
```

Comment:
Try writing a few tests to actually test the functionality of these functions as well.